src/main/java/org/opensearch/security/transport/SecurityRequestHandler.java

```diff
@@ -326,6 +327,11 @@ protected void addAdditionalContextValues(final String action, final TransportRe
             }
         }
 
+        String extensionUniqueId = getThreadContext().getHeader("extension_unique_id");
+        if (extensionUniqueId != null) {
```

Comment:
This check to also ensure that this `extension_unique_id` is from a registered extension within the `ExtensionsManager`.